packages/node/src/handlers.ts

```diff
@@ -315,6 +316,48 @@ export function errorHandler(options?: {
   };
 }
 
+interface SentryTrpcMiddlewareOptions {
+  attachRpcInput?: boolean;
+}
+
+interface TrpcMiddlewareArguments<T> {
+  path: string;
+  type: 'query' | 'mutation' | 'subscription';
+  next: () => T;
+  rawInput: unknown;
+}
+
+/**
+ * Sentry tRPC middleware that names the handling transaction after the called procedure.
+ *
+ * Use the Sentry tRPC middleware in combination with the Sentry server integration,
+ * e.g. Express Request Handlers or Next.js SDK.
+ */
+export async function trpcMiddleware(options: SentryTrpcMiddlewareOptions = {}) {
+  return function <T>({ path, type, next, rawInput }: TrpcMiddlewareArguments<T>): T {
+    const hub = getCurrentHub();
+    const clientOptions = hub.getClient()?.getOptions();
+    const sentryTransaction = hub.getScope()?.getTransaction();
+
+    if (sentryTransaction) {
+      sentryTransaction.setName(`trcp/${path}`, 'route');
+      sentryTransaction.op = 'rpc.server';
+
+      const trpcData: Record<string, unknown> = {
+        procedureType: type,
+      };
+
+      if (options.attachRpcInput !== undefined ? options.attachRpcInput : clientOptions?.sendDefaultPii) {
+        trpcData.procedureInput = normalize(rawInput);
+      }
+
+      sentryTransaction.setData('trpc', trpcData);
```

Comment:
`transaction.setData` does nothing on the event atm - can we add it as context instead? If we do that, can just just add it globally on the scope so that errors also get this info?

Reply:
I think we had this exact discussion before ^^ Data definitely appears in the transaction view:

![Screenshot 2023-03-30 at 09 52 49](https://user-images.githubusercontent.com/8118419/228767984-c2f7289f-de23-47ee-ac23-9e4321124946.png)

But I agree that a context is more valuable: 6ff778e7333e271d45b2644830b6c3f0f2740225

Reply:
Ah it does appear in trace context!!

Keep forgetting 🤦